### PR TITLE
[Aten] Fix XPU convolution_overrideable input memory format.

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124841

[Aten] Fix convolution_overrideable input memory format.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]
Any test cases?

**Comment by etaf** [comment]
> Any test cases?

Test case added.

File: test/xpu/test_conv.py [hunk 1]
```diff
@@ -1240,6 +1241,30 @@ def test_conv_double_backward_strided_with_3D_input_and_weight(self, device):
         self.assertEqual(grad_input.shape, input.shape)
         self.assertEqual(grad_weight.shape, weight.shape)
 
+    def test_channels_last_ouput_stride(self, device):
+        dtype = torch.float32
```
Comment (EikanWang) [inline test/xpu/test_conv.py:1245]: We should verify different data types.

File: test/xpu/test_conv.py [hunk 2]
```diff
@@ -1240,6 +1241,30 @@ def test_conv_double_backward_strided_with_3D_input_and_weight(self, device):
...
+            return out
+
+        # input NHWC, output NHWC
+        out = get_conv_out(device, torch.float32)
+        assert_size_stride(out, (2, 512, 7, 7), (25088, 1, 3584, 512))
+
+        # Like most conv backend, xpu does not support float64 for chanel last conv.
+        # input NHWC, output NCHW
+        out = get_conv_out(device, torch.float64)
+        assert_size_stride(out, (2, 512, 7, 7), (25088, 49, 7, 1))
```
Comment (EikanWang) [inline test/xpu/test_conv.py:1266]: It is xpu-specific. Please separate it from current case and add a decorator - `@onXPUOnly`

**Review by EikanWang — CHANGES_REQUESTED** [review]

**Comment by EikanWang** [comment]
@etaf , could you please check how to enhance ops to add such cases?

**Review by jgong5 — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@malfet , @albanD , @atalman , any comments?

**Comment by albanD** [comment]
I'm not sure what the contract with backend_memory_format and convolution_overrideable is tbh. @jbschlosser do you know?
I would expect that the most generic thing we do here is to pass in the inputs as-is and if your implementation has contraints on the contiguousness, you should call .contiguous() inside your kernel.
The problem with the proposal here is that if your kernel was effectively handling non-contiguous inputs. We're introducing and extra full copy that you cannot opt-out of.

**Comment by EikanWang** [comment]
The initial concept is that aligning with the backend-specific memory format may be a common practice for performance reasons. Thank you, @albanD. We will enhance the PR by transferring the modification to XPU-Conv.

**Review by EikanWang — APPROVED** [review]
LGTM.

**Review by albanD — APPROVED** [review]
Thanks for the update, SGTM

**Pull request closed without merging**